tests/pipelines/test_pipelines_common.py

```diff
@@ -199,6 +199,38 @@ def test_unbatch_attentions_hidden_states(self):
         outputs = text_classifier(["This is great !"] * 20, batch_size=32)
         self.assertEqual(len(outputs), 20)
 
+    @require_torch
+    def test_torch_dtype_property(self):
+        import torch
+
+        model_id = "hf-internal-testing/tiny-random-distilbert"
+
+        # If dtype is specified in the pipeline constructor, the property should return that type
+        pipe = pipeline(model=model_id, torch_dtype=torch.float16)
+        self.assertEqual(pipe.torch_dtype, torch.float16)
+
+        # If the underlying model changes dtype, the property should return the new type
+        pipe.model.to(torch.bfloat16)
+        self.assertEqual(pipe.torch_dtype, torch.bfloat16)
+
+        # Even if the model dtype is the default one, we can safely assume the pipeline supports torch_dtype
+        # as it is constructed with torch_dtype specified
+        pipe.model.to(torch.float32)
+        self.assertEqual(pipe.torch_dtype, torch.float32)
+
+        # If dtype is NOT specified in the pipeline constructor, the property should NOT return type
+        # as we don't know if the pipeline supports torch_dtype
```

Comment:
I don't really agree here, we should always return the dtype of the model just for consistence. The pipeline should error out normally and we should not assume that None torch_dtype == not supported

Reply:
I agree with the consistency concern, but also wonder that the model's dtype doesn't always translate to **torch**_dtype, like a model loaded with tensorflow/jax backend. If that sounds ok, I will proceed with simply propagating the model dtype:)

Reply:
If you have a `FlaxLlamaModel` then accessing the dtype should just be consistent imo